src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -5715,6 +5715,41 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             // Push string and substring argument on top of Vector (or Function Arguments Stack basically)
             fn_args.push_back(al, str);
             fn_args.push_back(al, suffix);
+        } else if (attr_name == "partition") {
+
+            /*  
+                str.partition(seperator)        ---->
+
+                Split the string at the first occurrence of sep, and return a 3-tuple containing the part 
+                before the separator, the separator itself, and the part after the separator. 
+                If the separator is not found, return a 3-tuple containing the string itself, followed 
+                by two empty strings.
+            */
+            
+            if(args.size() != 1) {
+                throw SemanticError("str.partition() takes one argument",
+                        loc);
+            }
+
+            ASR::expr_t *arg_seperator = args[0].m_value;
+            ASR::ttype_t *arg_seperator_type = ASRUtils::expr_type(arg_seperator);
+            if (!ASRUtils::is_character(*arg_seperator_type)) {
+                throw SemanticError("str.partition() takes one argument of type: str",
+                        loc);
+            }
+
```

Comment:
Add this check here as well:
```cpp
            if(s_var.size() == 0) {
                 throw SemanticError("string to undergo partition cannot be empty",loc);
             }
```

Reply:
But it is not possible to do that here, because s_var is an expr_t whose value is not known until compile time and gets known only after the call to method is made? What do you think?

Reply:
I don't we can do it directly here as s_var is a `expr_t` @Thirumalai-Shaktivel